src/ctapipe/utils/astro.py

```diff
@@ -4,63 +4,199 @@
-    ~6.5, while the faintest included star has mag=7.96. :cite:p:`bright-star-catalog`
+def select_stars(stars, pointing=None, radius=None, magnitude_cut=None, band="B"):
+    """
+    utility function to cut stars based on magnitude and/or location
 
     Parameters
     ----------
+    stars: astropy table
+        Table of stars, including magnitude and coordinates
     pointing: astropy Skycoord
-       pointing direction in the sky (if none is given, full sky is returned)
+        pointing direction in the sky (if none is given, full sky is returned)
     radius: astropy angular units
-       Radius of the sky region around pointing position. Default: full sky
+        Radius of the sky region around pointing position. Default: full sky
     magnitude_cut: float
-        Return only stars above a given magnitude. Default: None (all entries)
+        Return only stars above a given apparent magnitude. Default: None (all entries)
+    band: string
+        wavelength band to use for the magnitude cut options are V and B. Default: 'B' (all entries)
 
     Returns
     -------
     Astropy table:
-       List of all stars after cuts with names, catalog numbers, magnitudes,
-       and coordinates
+       List of all stars after cuts with same keys as the input table stars
     """
-    from ctapipe.utils import get_table_dataset
-
-    catalog = get_table_dataset("yale_bright_star_catalog5", role="bright star catalog")
-
-    starpositions = SkyCoord(
-        ra=Angle(catalog["RAJ2000"], unit=u.deg),
-        dec=Angle(catalog["DEJ2000"], unit=u.deg),
-        frame="icrs",
-        copy=False,
-    )
-    catalog["ra_dec"] = starpositions
 
     if magnitude_cut is not None:
-        catalog = catalog[catalog["Vmag"] < magnitude_cut]
+        if band == "B":
+            if "Bmag" in stars.keys():
+                stars = stars[stars["Bmag"] < magnitude_cut]
+            elif "BTmag" in stars.keys():
+                stars = stars[stars["BTmag"] < magnitude_cut]
+            else:
+                raise ValueError(
+                    "The requested catalogue has no compatible magnitude for the B band"
+                )
+
+        if band == "V":
+            if "Vmag" in stars.keys():
+                stars = stars[stars["Vmag"] < magnitude_cut]
+            elif "VTmag" in stars.keys():
+                stars = stars[stars["VTmag"] < magnitude_cut]
+            else:
+                raise ValueError(
+                    "The requested catalogue has no compatible magnitude for the V band"
+                )
 
     if radius is not None:
         if pointing is None:
             raise ValueError(
                 "Sky pointing, pointing=SkyCoord(), must be "
                 "provided if radius is given."
             )
-        separations = catalog["ra_dec"].separation(pointing)
-        catalog["separation"] = separations
-        catalog = catalog[separations < radius]
+        separations = stars["ra_dec"].separation(pointing)
```

Comment:
Selecting stars by separation using ra dec only makes sense after applying space motion, so I think we really need to apply it inside the `get...` functions.

Reply:
i put that into both get_stars functions. It should now cut on location after motion is applied. The user now needs to specify the time for the lookup